### PR TITLE
wip directions on emphasizing user/workspace first

### PR DESCRIPTION
**Pull request by tildeee**
WIP for this trello card https://trello.com/c/PQFOQEGg

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,13 +73,59 @@ For this project, we have provided our standard test infrastructure:
...
+
+We want every project to have a `Workspace` class. To setup this class, do the following:
+
+1. Create `lib/workspace.rb` and create a class named `Workspace`
+1. The `initialize` should not take in any parameters
+1. Within `initialize`, the class should set up the instance variables `@users` and `@channels`. Set the value of `@users` to an empty array. Set the value of `@channels` to an empty array. Eventually, these should be refactored so that they are arrays of users or arrays of channels
+1. Set `attr_reader` for these two instance variables in `Workspace`
+1. Run `$ ruby lib/slack.rb` to make sure that the Ruby program runs without syntax errors. You probably should get
+1. With your pair, determine: What is the relationship between `Workspace` and a `User`? Is it inheritance or composition? Is-a or has-a or has-many? How do we know from the Ruby code?
+1. With your pair, determine: What is the relationship between `Workspace` and a `Channel`? Is it inheritance or composition? Is-a or has-a or has-many? How do we know from the Ruby code?
```
Comment (CheezItMan) [inline README.md:101]: Maybe ask them to draw the relationships with a UML diagram, if it helps.

**Review by CheezItMan — COMMENTED** [review]
Definitely unfinished, but so far so good.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,6 @@
 /coverage/
 
+*/.DS_Store
```
Comment (kaidamasaki) [inline .gitignore]: I'd recommend writing this as:
```suggestion
.DS_Store
```

File: design_activity.md [hunk 3]
```diff
@@ -10,19 +10,20 @@ While we encourage you to use the provided design for this project, there is muc
 - Compare and contrast different designs for the same program
 - Implement a program from a provided UML diagram
 
-## Your Design
+## First Draft Design
 
-Read through the project requirements with your partner, then follow our standard [object-oriented design process](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/02-intermediate-ruby/oo-design.md).
+1. Read through the project requirements with your partner, then follow our standard [object-oriented design process](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/02-intermediate-ruby/oo-design.md).
+    - What are the nouns, and how are they composed into objects and relations?
```
Comment (kaidamasaki) [inline design_activity.md]: Do we want to include a question about what properties the nouns have and/or split the relation and object parts into two sub bullets?

File: design_activity.md [hunk 4]
```diff
@@ -10,19 +10,20 @@ While we encourage you to use the provided design for this project, there is muc
...
+    - What are the verbs, and which verb belongs with which object?
+2. From above, we are leading you to certain classes and dependencies. We want `lib/slack.rb` to have only **one dependency**: `Workspace`. We want `Workspace` to have at least two dependencies: `User` and `Channel`. We want `Workspace` to have many `User`s (composition). Also, we want `Workspace` to have many `Channel`s (composition). After reading through the requirements, what verbs are associated with `User`? What verbs are associated with `Channel`? What's **similar**? What's **different**?
```
Comment (kaidamasaki) [inline design_activity.md]:
```suggestion
2. From above, we are leading you to certain classes and dependencies. We want `lib/slack.rb` to have only **one dependency**: `Workspace`. We want `Workspace` to have at least two dependencies: `User` and `Channel`. We want `Workspace` to have many `User`s (composition). Also, we want `Workspace` to have many `Channel`s (composition). After reading through the requirements:
  - What verbs are associated with `User`? 
  - What verbs are associated with `Channel`? 
  - What's **similar**? 
  - What's **different**?
```

File: design_activity.md [hunk 5]
```diff
@@ -10,19 +10,20 @@ While we encourage you to use the provided design for this project, there is muc
...
+3. The `User` and `Channel` classes have similar enough verbs and responsibilities, that we could design them to have inheritance relationships. What ideas can you come up with? What would the common behaviors/methods be named?
```
Comment (kaidamasaki) [inline design_activity.md]:
```suggestion
3. The `User` and `Channel` classes have similar enough verbs and responsibilities, that we could design them to have inheritance relationships. 
  - What ideas can you come up with? 
  - What would the common behaviors/methods be named?
```

Comment (kaidamasaki) [reply design_activity.md]: Based on this paragraph I feel like there is a chance that we will get `class User < Channel`.  Do we want to tweak the wording to suggest a new parent class?

File: design_activity.md [hunk 6]
```diff
@@ -32,19 +33,37 @@ Note that only public methods and attributes are shown.
...
+
+`slack.rb` should only interact with one class, `Workspace`.
+
+We want to minimize the number of classes it depends on if possible.
+
+The `slack.rb` file should not create, call, or use the `User`, `Channel`, or `Recipient` classes at all. The `slack.rb` file should mainly be interacting with the instance of `Workspace` created in the line `workspace = Workspace.new`. All `User`-related information that `slack.rb` receives should be returned from calling a method defined in the `Workspace` class.
+
+</details>
+
+### What If I Don't Understand The Design?
```
Comment (kaidamasaki) [inline design_activity.md:68]: Consider adding a plug for `#classroom-support`.

**Review by kaidamasaki — COMMENTED** [review]
Generally looks good.  A few small things though.

File: design_activity.md (hunk 4 above)
Comment (tildeee) [reply design_activity.md]: As a note for the future:

For GitHub Markdown formatting,

if you're mixing list style types (numbered vs. bulleted)
You need to double indent in the nested list.

I'm fixing this in a commit now but in the future that's what'll make it look good!

**Pull request merged**